src/compiler/checker.ts

```diff
@@ -25033,15 +25039,19 @@ namespace ts {
             }
 
             if (entityName.parent.kind === SyntaxKind.ExportAssignment && isEntityNameExpression(entityName)) {
-                return resolveEntityName(entityName,
-                    /*all meanings*/ SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace | SymbolFlags.Alias);
+                // Even an entity name expression that doesn't resolve as an entityname may still typecheck as a property access expression
+                const success = resolveEntityName(entityName,
+                    /*all meanings*/ SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace | SymbolFlags.Alias, /*ignoreErrors*/ true);
+                if (success && success !== unknownSymbol) {

```

Comment:
Does this mean `resolveEntityName` is inconsistent about its return value? Could we just fix that function to use only one of `undefined` or `unknownSymbol`?

Reply:
I believe it can become `unknownSymbol` is it resolves to an alias which was circular, which gives `unknownSymbol` once the circularity is terminated; however we'd still like to get a real symbol associated with non-alias meanings of the name if possible, since `unknownSymbol` is fairly useless (and the `checkExpression` bit accomplishes that).